core/state_transition.go

```diff
@@ -730,6 +744,18 @@ func (st *stateTransition) refundGas(refundQuotient uint64) uint64 {
 		st.evm.Config.Tracer.OnGasChange(st.gasRemaining, st.gasRemaining+refund, tracing.GasChangeTxRefunds)
 	}
 
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && !st.msg.IsDepositTx && isIsthmus {
+		// Return ETH to transaction sender for operator cost overcharge.
+		operatorCostGasLimit := st.evm.Context.OperatorCostFunc(st.msg.GasLimit, st.evm.Context.Time)
+		operatorCostGasUsed := st.evm.Context.OperatorCostFunc(st.gasUsed(), st.evm.Context.Time)
```

Comment:
See other comment, noting that the value of `st.gasUsed()` will potentially be different (higher) here than in line 664, because we apply state refunds later. Just flagging to make sure this is intended.

Reply:
This aligns now with the Rust implementation. Needs to be covered with a test. https://github.com/ethereum-optimism/op-geth/issues/528